be/src/vec/functions/function_date_or_datetime_computation.h

```diff
@@ -580,10 +617,72 @@ class FunctionDateOrDateTimeComputation : public IFunction {
             // vector-const
             if (const auto* nest_col1_const = check_and_get_column<ColumnConst>(*nest_col1)) {
                 rconst = true;
-                const auto col1_inside_const =
-                        assert_cast<const IntervalColumnType&>(nest_col1_const->get_data_column());
-                Op::vector_constant(sources->get_data(), res_col->get_data(),
-                                    col1_inside_const.get_data()[0], nullmap0, nullmap1);
+                if constexpr (Transform::IntervalPRealType == TYPE_STRING) {
```

Comment:
type is `TYPE_STRING` could represent `day_second` just now. shouldn't put the logic here.